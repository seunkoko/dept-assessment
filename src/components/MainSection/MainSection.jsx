// eslint-disable-next-line
import React from 'react';

import FilterDropdown from '../FilterDropdown/FilterDropdown';
import NewsSection from '../NewsSection/NewsSection';
import TopAd from '../AdSection/TopAd';
import BottomAd from '../AdSection/BottomAd';
import Quote from '../common/Quote';
import ClientSection from '../ClientSection/ClientSection';
import MessageSection from '../MessageSection/MessageSection';
import Footer from '../Footer/Footer';

import './MainSection.scss';

import { workFilter, industryFilter } from '../../utils/filterData'
import {
  firstNewsData,
  secondNewsData,
  thirdNewsData,
  fourthNewsData,
} from '../../utils/news';


const MainSection = ({
  email,
  name,
  message,
  emailError,
  nameError,
  messageError,
  successMessage,
  handleAlertClose,
  handleOnChange,
  handleSubmitMessage,
}) => {
	return (
    <>
      <div className="container-fluid main-section-style">

        {/* Filter Section */}
        <div className="filter-style">
          <div>
            <p>Show me</p>
            <FilterDropdown
              sectionId={"all-work"}
              displayName={"all work"}
              data={workFilter}
            />
          </div>

          <div className="mr-0">
            <p>in</p>
            <FilterDropdown
              sectionId={"all-industries"}
              displayName={"all industries"}
              data={industryFilter}
            />
          </div>
        </div>

        {/* First News Section */}
        <NewsSection
          data={firstNewsData}
        />

        {/* Top Ad Section */}
        <TopAd />

        {/* Second News Section */}
        <NewsSection
          data={secondNewsData}
        />

        {/* Bottom Ad Section */}
        <BottomAd />

        {/* Third News Section */}
        <NewsSection
          data={thirdNewsData}
        />

        {/* Quote */}
        <Quote />

        {/* Fourth News Section */}
        <NewsSection
          data={fourthNewsData}
        />

        {/* Client Section */}
        <ClientSection />

        {/* Message Section */}
        <MessageSection
          email={email}
          name={name}
          message={message}
          emailError={emailError}
          nameError={nameError}
          messageError={messageError}
          successMessage={successMessage}
          handleAlertClose={handleAlertClose}
          handleOnChange={handleOnChange}
          handleSubmitMessage={handleSubmitMessage}
        />

        {/* Footer */}
        <Footer />
      </div>
    </>
	);
};

export default MainSection;
