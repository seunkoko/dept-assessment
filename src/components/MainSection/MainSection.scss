@import "../../stylesheets/variables";

.main-section-style {
  margin: 30px auto;
  padding: 0;
  height: 300px;
  width: 94%;

  @media (max-width: $phone-width) {
    margin: 20px auto;
    width: 90%;
	}
}

.filter-style {
  width: 100%;
  text-align: right;

  > div {
    font-family: 'Teko-Regular';
    font-size: $medium-font;
    color: $primary-gray;
    margin-right: 10px;
    display: inline-block;

    > p {
      display: inline-block;
      margin-right: 6px;
    }

    @media (max-width: $phone-width) {
      display: flex;
    }
  }

  @media (max-width: $phone-width) {
    text-align: left;
  }
}