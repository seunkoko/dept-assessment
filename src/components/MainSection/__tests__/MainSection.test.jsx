import React from 'react';

import { shallow } from 'enzyme';

import MainSection from '../MainSection';

describe('MainSection', () => {
	const component = shallow(<MainSection />);

	it('should render MainSection component', () => {
		expect(component).toMatchSnapshot();
  });
  
});