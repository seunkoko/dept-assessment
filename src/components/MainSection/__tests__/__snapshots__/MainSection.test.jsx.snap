// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainSection should render MainSection component 1`] = `
<Fragment>
  <div
    className="container-fluid main-section-style"
  >
    <div
      className="filter-style"
    >
      <div>
        <p>
          Show me
        </p>
        <FilterDropdown
          data={
            Array [
              Object {
                "id": 1,
                "name": "Frontend Engineer",
              },
              Object {
                "id": 2,
                "name": "Backend Engineer",
              },
              Object {
                "id": 3,
                "name": "Data Scientist",
              },
            ]
          }
          displayName="all work"
          sectionId="all-work"
        />
      </div>
      <div
        className="mr-0"
      >
        <p>
          in
        </p>
        <FilterDropdown
          data={
            Array [
              Object {
                "id": 1,
                "name": "E-Commerce",
              },
              Object {
                "id": 2,
                "name": "Medicine",
              },
              Object {
                "id": 3,
                "name": "Robotics",
              },
            ]
          }
          displayName="all industries"
          sectionId="all-industries"
        />
      </div>
    </div>
    <NewsSection
      data={
        Array [
          Object {
            "description": "A summer island in the Netherlands",
            "id": "bolcom",
            "image": "test-file-stub",
            "moreDescription": "Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident.",
            "title": "BOLCOM",
          },
          Object {
            "description": "Not some average banking website",
            "id": "kempen",
            "image": "test-file-stub",
            "moreDescription": "Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident.",
            "title": "KEMPEN",
          },
          Object {
            "description": "Beautiful design meets innovative technology",
            "id": "phillips",
            "image": "test-file-stub",
            "moreDescription": "Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident.",
            "title": "PHILIPS",
          },
          Object {
            "description": "A hundred years of Mondriaan & De Stijl",
            "id": "gemeentemuseum",
            "image": "test-file-stub",
            "moreDescription": "Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident.",
            "title": "GEMEENTEMUSEUM",
          },
        ]
      }
    />
    <TopAd />
    <NewsSection
      data={
        Array [
          Object {
            "description": "Delivering clarity on a global scale",
            "id": "lightning",
            "image": "test-file-stub",
            "moreDescription": "Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident.",
            "title": "BE LIGHTNING",
          },
          Object {
            "description": "Swipe to find your next holiday destination",
            "id": "tui",
            "image": "test-file-stub",
            "moreDescription": "Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident.",
            "title": "TUI",
          },
        ]
      }
    />
    <BottomAd />
    <NewsSection
      data={
        Array [
          Object {
            "description": "A campaign for the limited edition letter packs",
            "id": "chocomel",
            "image": "test-file-stub",
            "moreDescription": "Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident.",
            "title": "CHOCOMEL",
          },
          Object {
            "description": "Live like a champion with Jerome Booteng",
            "id": "jbl",
            "image": "test-file-stub",
            "moreDescription": "Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident.",
            "title": "JBL",
          },
          Object {
            "description": "Innovative SEO and content strategy for Zalando",
            "id": "zalando",
            "image": "test-file-stub",
            "moreDescription": "Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident.",
            "title": "ZALANDO",
          },
          Object {
            "description": "The search for the most influential book ever",
            "id": "konin",
            "image": "test-file-stub",
            "moreDescription": "Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident.",
            "title": "KONINKLIJKE BIBLIOTHEEK",
          },
        ]
      }
    />
    <Quote />
    <NewsSection
      data={
        Array [
          Object {
            "description": "Delivering complex commerce solutions",
            "id": "liberty",
            "image": "test-file-stub",
            "moreDescription": "Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident.",
            "title": "LIBERTY GLOBAL",
          },
          Object {
            "description": "Swipe to find your next holiday destination",
            "id": "arla",
            "image": "test-file-stub",
            "moreDescription": "Anim pariatur cliche reprehenderit, enim eiusmod high life accusamus terry richardson ad squid. Nihil anim keffiyeh helvetica, craft beer labore wes anderson cred nesciunt sapiente ea proident.",
            "title": "ARLA",
          },
        ]
      }
    />
    <ClientSection />
    <MessageSection />
    <Footer />
  </div>
</Fragment>
`;
